File: 백준/Silver/18111. 마인크래프트/마인크래프트.js
const input = require('fs').readFileSync(0).toString().trim().split(/\s+/).map(Number);
const N = input.shift();
const M = input.shift();
const B = input.shift();
const blocks = input;

let minTime = Infinity;
let result = 0;
const minH = Math.min(...blocks);
const maxH = Math.max(...blocks);

for (let h = minH; h <= maxH; h++) {
    let remove = 0;
    let add = 0;
    
    for (const block of blocks) {
        if (block > h) {
            remove += block - h;
        } else if (block < h) {
            add += h - block;
        }
    }
    
    if ((B + remove) >= add) {
        const time = remove * 2 + add;
        if (time < minTime) {
            minTime = time;
            result = h;
        } else if (time === minTime) {
            result = h;
        }
    }
}

console.log(minTime, result);

File: 백준/Silver/18111. 마인크래프트/마인크래프트.py
import sys

data = sys.stdin.read().split()
idx = 0
N, M, B = int(data[0]), int(data[1]), int(data[2])
blocks = [int(data[i]) for i in range(3, 3 + N*M)]

min_time = float('inf')
result_h = 0
min_h = min(blocks)
max_h = max(blocks)

for h in range(min_h, max_h + 1):
    remove = 0
    add = 0
    
    for block in blocks:
        if block > h:
            remove += block - h
        elif block < h:
            add += h - block
            
    if B + remove >= add:
        time = remove * 2 + add
        if time < min_time:
            min_time = time
            result_h = h
        elif time == min_time:
            result_h = h

print(min_time, result_h)